driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -21,4 +21,37 @@
  */
 public interface NumberExpression extends Expression {
 
+    NumberExpression multiply(NumberExpression n);
+
+    default NumberExpression multiply(final double multiply) {
```

Comment:
@jyemin raised an issue with `double` causing unexpected promotion of underlying BSON type causing a ClassCast during:

```
        long d1 = (long) result.get("d");
        long d2 = result.getLong("d");
        long d3 = result.get("d", Long.class);
```

Fixed in:

https://github.com/mongodb/mongo-java-driver/pull/1052/files#diff-276193793ed4e0294e212d9927de707b2d000804cbf649f2a0d3888e2ec9adf6L26

Reply:
I pulled this in to this PR